Client/Frontend/Home/ActivityStreamPanel.swift

```diff
@@ -279,20 +297,26 @@ extension ActivityStreamPanel {
         guard let host = siteURL.normalizedHost() else {
             return
         }
+        // if the default top sites contains the siteurl. also wipe it from default suggested sites.
+        if defaultTopSites().filter({$0.tileURL.absoluteString != siteURL.absoluteString}).isEmpty == false {
+            deleteTileForSuggestedSite(siteURL.absoluteString)
+        }
         profile.history.removeHostFromTopSites(host).uponQueue(dispatch_get_main_queue()) { result in
             guard result.isSuccess else { return }
             self.reloadTopSites()
         }
     }
 
-    private func siteToItem(site: Site?) -> TopSiteItem? {
-        guard let site = site else {
-            return nil
-        }
-        guard let faviconURL = site.icon?.url else {
-            return TopSiteItem(urlTitle: site.tileURL.extractDomainName(), faviconURL: nil, siteURL: site.tileURL)
-        }
-        return TopSiteItem(urlTitle: site.tileURL.extractDomainName(), faviconURL: NSURL(string: faviconURL)!, siteURL: site.tileURL)
+    private func deleteTileForSuggestedSite(siteURL: String) {
+        var deletedSuggestedSites = profile.prefs.arrayForKey(DefaultSuggestedSitesKey) as? [String] ?? []
+        deletedSuggestedSites.append(siteURL)
+        profile.prefs.setObject(deletedSuggestedSites, forKey: DefaultSuggestedSitesKey)
+    }
+
+    private func defaultTopSites() -> [Site] {
```

Comment:
Does this need to be a method that gets recalculate each time instead of caching the value?